### PR TITLE
[5.3] Fix match only APP_KEY in replace

### LINKED ISSUE #17140: PUSHER_APP_KEY .env
- Laravel Version: 5.4 dev
- PHP Version: 7.1
- Database Driver & Version:

### Description:
It appears that the new default .env file containing APP_KEY and PUSHER_APP_KEY gets a double generated APP_KEY upon installation.

### Steps To Reproduce:
`laravel new demo --dev`
See .env file containing: 

```
APP_KEY=base64:xWiu0LfiJsvKXxzDq6YgqL4mPqd3SapixvYkuRbT2cw=
# [redacted]
PUSHER_APP_KEY=base64:xWiu0LfiJsvKXxzDq6YgqL4mPqd3SapixvYkuRbT2cw=
```

### PR DESCRIPTION
**Pull request by TheoKouzelis**
Changed str_replace to preg_replace to ensure no clashes with other environment variables.

---

Closes #17140.

### CODE REVIEW

File: src/Illuminate/Foundation/Console/KeyGenerateCommand.php [hunk 1]
```diff
@@ -69,4 +69,16 @@ protected function generateRandomKey()
...
+     * Get a regex pattern that will match env APP_KEY with any random key.
+     *
+     * @return string
+     */
+    protected function getAppKeyPattern()
+    {
+        $escapedKey = preg_quote('='.$this->laravel['config']['app.key'], '/');
+
+        return "/^APP_KEY$escapedKey/m";
+    }
```
Comment (tillkruss) [inline src/Illuminate/Foundation/Console/KeyGenerateCommand.php]: This does not read nice to me 😕

Comment (TheoKouzelis) [inline src/Illuminate/Foundation/Console/KeyGenerateCommand.php]: Is there anything you would recommend to make it read better?

Comment (tillkruss) [inline src/Illuminate/Foundation/Console/KeyGenerateCommand.php]: I made a quick PR: https://github.com/laravel/framework/pull/17226

Comment (TheoKouzelis) [inline src/Illuminate/Foundation/Console/KeyGenerateCommand.php]: Thanks for the PR and the code review.

**Pull request merged**